testdata/inMetrics_ExportMetricsOfAllTypes.txt

```diff
@@ -0,0 +1,332 @@
+metric_descriptor: <
+  name: "ocagent.io/metricCummDouble"
+  description: "Description of metricCummDouble"
+  unit: "ms"
+  type: CUMULATIVE_DOUBLE
+  label_keys: <
+    key: "empty_key"
+  >
+  label_keys: <
+    key: "operation_type"
+  >
+>
+timeseries: <
+  start_timestamp: <
+    seconds: 1543160298
+    nanos: 100000090
+  >
+  label_values: <
+    has_value: true
+  >
+  label_values: <
+    value: "test_1"
+    has_value: true
+  >
+  points: <
+    timestamp: <
+      seconds: 1543160298
+      nanos: 100000997
+    >
+    double_value: 35.5
+  >
+>
+---
```

Comment:
Do you need these separators? I think you can use a metric list instead, something like
metric: <
  metric_descriptor: <...>
  timeseries: <...>
>
metric: <
  metric_descriptor: <...>
  timeseries: <...>
>
...

Then in test.go:
```go
var metrics []*metricspb.metric
proto.UnmarshallTxt(metrics, file)
```

Reply:
Same for metric descriptor requests and time series requests.

Reply:
UnmarshalText only accepts Message
```go
func UnmarshalText(s string, pb Message) error {
```
I would have to create wrapper proto like
```
message MetricList {
  repeated opencensus.proto.metric.v1.Metric metric = 1;
}
```
Let me know if I missed something.

Reply:
Got it - I didn't know that, LGTM then.